rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -81,663 +88,747 @@
-		 * 
-		 * @param pn
-		 *            the deploying node to update
-		 * @param newDescription
-		 *            the new description
-		 */
-		protected abstract void setDescription(RMDeployingNode pn, String newDescription);
-
-		/**
-		 * To update the lost field of the deploying node
-		 * 
-		 * @param pn
-		 *            The deploying node to update
-		 */
-		protected abstract void setLost(RMDeployingNode pn);
-	}
+    /** class' logger */
+    protected static final Logger logger = Logger.getLogger(InfrastructureManager.class);
+
+    /** manager's node source */
+    protected NodeSource nodeSource;
+
+    /** deploying nodes list */
+    private final Map<String, RMDeployingNode> deployingNodes = new ConcurrentHashMap<>();
+
+    private final Map<String, RMDeployingNode> lostNodes = new ConcurrentHashMap<>();
+
+    /**
+     * node list, miror of nodesource.getAliveNodes(), to implement random
+     * access
+     */
+    private Hashtable<String, Node> acquiredNodes = new Hashtable<>();
+
+    private final ReentrantLock nodeAcquisitionLock = new ReentrantLock();
+
+    // shared fields, needs to be volatile not to cache it
+    private volatile boolean usingDeployingNodes = false;
+
+    // shutdown marker
+    private volatile boolean shutdown = false;
+
+    // used to timeout the nodes
+    private transient Timer timeouter = null;
+
+    protected String rmUrl;
+
+    public InfrastructureManager() {
+    }
+
+    /**
+     * Sets an infrastructure node source. Also sets the field rmUrl if it is
+     * not provided by users.
+     *
+     * @param nodeSource
+     *            policy node source
+     */
+    public void setNodeSource(NodeSource nodeSource) {
+        this.nodeSource = nodeSource;
+        this.rmUrl = nodeSource.getRegistrationURL();
+    }
+
+    /**
+     * To retrieve nodes whose registration status is deploying or lost.
+     *
+     * @return nodes whose registration status is deploying or lost.
+     */
+    public ArrayList<RMDeployingNode> getDeployingNodes() {
+        ArrayList<RMDeployingNode> result;
+        synchronized (deployingNodes) {
+            Collection<RMDeployingNode> rmDeployingNodes = this.deployingNodes.values();
+            Collection<RMDeployingNode> rmLostNodes = this.lostNodes.values();
+            result = new ArrayList<>(rmDeployingNodes.size() + rmLostNodes.size());
+            result.addAll(rmDeployingNodes);
+            result.addAll(rmLostNodes);
+        }
+        return result;
+    }
+
+    public RMDeployingNode getDeployingNode(String nodeUrl) {
+        RMDeployingNode deployingNode;
+
+        synchronized (deployingNodes) {
+            deployingNode = deployingNodes.get(nodeUrl);
+        }
+
+        if (deployingNode == null) {
+            return lostNodes.get(nodeUrl);
+        }
+
+        return deployingNode;
+    }
+
+    /**
+     * To remove a deploying node given its url
+     *
+     * @param pnUrl
```

Comment:
To be honest, I don't know the meaning of _pnUrl_. Perhaps, persistent node URL? You see changes on these lines because of the new formatting rules applied on my IDE but not yet fully on the Scheduling project.